tools/prepare_python_system_tests.py

```diff
@@ -7,20 +7,6 @@
 import urllib.request
 import zipfile
 
-parser = argparse.ArgumentParser(description='Downloads the latest release artifacts from nimi-python and runs system tests on the specified python package.',
-                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument('-d', '--driver', required=True, type=str,
-                    help='Python package name.',
-                    choices=os.listdir('src'))
-parser.add_argument('-pv', '--python-version', required=False, type=str,
-                    help='Python version to be run. This is used to invoke the appropriate tox environment.',
-                    choices=['py35', 'py36', 'py37', 'py38', ], default='py38')
-parser.add_argument('-pb', '--python-bitness', required=False, type=str,
-                    help='Python bitness to be run. "32" means pass "--32" to tox, which will force 32 bit. "any" does not pass anything to tox, so it will used whatever bitness is installed, preferring 64 if available',
-                    choices=['32', 'any'], default=None)
-args = parser.parse_args()
-
-
 print('****Installing tox to Python.****')
```

Comment:
Doesn't this mean that this script will now print and do a bunch of stuff before processing command line arguments?

You want to validate all inputs before doing anything. Why was it moved down in the script?

Reply:
Without this change, the script would throw FileNotFoundError as it would only download the 'prepare_python_system_test'. There won't be any 'src' folder. Due to this, the package has to be downloaded first

Reply:
As part of providing help for `-d` option, script lists the available choices:
```
parser.add_argument('-d', '--driver', required=True, type=str,
                    help='Python package name.',
                    choices=os.listdir(src))
```

I suggested reordering so we can avoid hard-coding the choices. We could remove the prints but still do the processing upfront. If you think that's overkill, we could just get rid of the choices.

Reply:
That is an unfortunate chicken and egg problem. I think the root cause is that the scripts aren't decoupled from each other.

There are two scripts:

1. prepare_python_system_tests.py
2. run_python_system_tests.py

The first one prepares for, then runs the selected system test. This is not what the name of the script conveys. It calls on the second script in order to do it.

What if you decoupled the two scripts so that they are more specific and do just the one thing?

The first script prepares to run tests, the second one runs the specified test. It means the test suites need to call two scripts, but it also means that you could in theory test several drivers (for example, NI-SCOPE and NI-TClk) on the same Test Suite which would be desirable.

Reply:
I like this idea. @pmitra30, I'm working on a new PR to implement Marcos' idea.

Reply:
We can do it this way:

1. We can install the necessary modules and the latest package using `prepare_python_system_tests.py`. To do that we do not need to restrict the driver argument. Actually we do not need to pass any arguments to install the modules and download the nimi-python package.

2. In `run_python_system_tests.py` we can have the logic to pass the driver name and python version to run the tests for any driver